frontend/src/api/contexts/config/ConfigContext.tsx

```diff
@@ -0,0 +1,26 @@
+import { createContext, useContext, useEffect, useState } from 'react'
+import { HasChildren } from '../../../util/react-types.util'
+import axios from 'axios'
+import { API_BASE_URL } from '../../../util/configs/environment.config'
+import { ConfigDto } from './types'
+
+export const ConfigContext = createContext<ConfigDto | undefined>(undefined)
+
+export const ConfigProvider = ({ children }: HasChildren) => {
+  const [config, setConfig] = useState<ConfigDto>()
+  const [loading, setLoading] = useState<boolean>(true)
+  useEffect(() => {
+    axios
+      .get<ConfigDto>(API_BASE_URL + '/api/app')
+      .then((res) => {
+        setConfig(res.data)
+      })
+      .catch(() => console.error('Theme fetch failed!'))
+      .finally(() => setLoading(false))
+  }, [])
+  // TODO: add loading screen
+  if (loading) return null
```

Comment:
Instead of these useStates and the possible nullability of `config` used in other components, we could use the `useQuery` hook as used in the AuthContext. But this is alright for now, we can leave it alone for now and create a refactor issue in the repo for later.